homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -0,0 +1,359 @@
+"""
+Generic GeoRSS events service.
+
+Retrieves current events (typically incidents or alerts) in GeoRSS format, and
+shows information on events filtered by distance to the HA instance's location
+and grouped by category.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.geo_rss_events/
+"""
+
+import asyncio
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (STATE_UNKNOWN, CONF_SCAN_INTERVAL,
+                                 CONF_UNIT_OF_MEASUREMENT, CONF_NAME,
+                                 CONF_ICON)
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+from homeassistant.helpers.event import async_track_time_interval
+
+REQUIREMENTS = ['feedparser==5.2.1', 'haversine==0.4.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CATEGORIES = 'categories'
+CONF_RADIUS = 'radius'
+CONF_URL = 'url'
+
+DEFAULT_ICON = 'mdi:alert'
+DEFAULT_NAME = "Event Service"
+DEFAULT_RADIUS_IN_KM = 20.0
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=5)
+DEFAULT_UNIT_OF_MEASUREMENT = 'Events'
+
+DOMAIN = 'geo_rss_events'
+ENTITY_ID_FORMAT = 'sensor.' + DOMAIN + '_{}'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM): vol.Coerce(float),
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_ICON, default=DEFAULT_ICON): cv.icon,
+    vol.Optional(CONF_CATEGORIES, default=[]): vol.All(cv.ensure_list,
+                                                       [cv.string]),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT,
+                 default=DEFAULT_UNIT_OF_MEASUREMENT): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices,
+                         discovery_info=None):  # pragma: no cover
+    """Set up the GeoRSS component."""
+    # Grab location from config
+    home_latitude = hass.config.latitude
+    home_longitude = hass.config.longitude
+    url = config.get(CONF_URL)
+    radius_in_km = config.get(CONF_RADIUS)
+    name = config.get(CONF_NAME)
+    icon = config.get(CONF_ICON)
+    categories = config.get(CONF_CATEGORIES)
+    scan_interval = config.get(CONF_SCAN_INTERVAL) or DEFAULT_SCAN_INTERVAL
```

Comment:
Since we're using scan interval directly, my preference would be to just redefine it in the schema here, and supply the default in the schema.

https://github.com/home-assistant/home-assistant/blob/2dab239021cbbf265f584088f2fed06abd8c1475/homeassistant/helpers/config_validation.py#L402